service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -47,4 +48,20 @@ public static void writeCORSHeaders(
     responseHeaders.putIfAbsent(ACCESS_CONTROL_ALLOW_HEADERS, corsSupportProperties.allowHeaders());
     responseHeaders.putIfAbsent(ACCESS_CONTROL_MAX_AGE, corsSupportProperties.maxAge());
   }
+
+  public static Optional<String> getToken(Map<String, String> headers) {
+    return Utils.getTokenFromCookie(headers).or(() -> Utils.getTokenFromAuthorization(headers));
+  }
+
+  private static Optional<String> getTokenFromCookie(Map<String, String> headers) {
```

Comment:
I know this is moved code -- just wondering if we should use `RelayedHttpRequest.getCookies()` instead of trying to parse the cookie value from the headers. Wondering if there are checks for expiration, etc that we're missing here..

Reply:
I thought I saw it googling but now opened Intellij and don't see it either in `com.microsoft.azure.relay` classes. The current approach is probably the right thing to do then.